tests/tests.go

```diff
@@ -306,10 +308,38 @@ func addLabelsToNamespace(namespace string, labels map[string]string) error {
 	return err
 }
 
+// function checks what is the currently configured opt-mode configured in a specific webhook
+func getOptMode(webhookName string) (string, error) {
+	mutatingWebhook, err := testClient.KubeClient.AdmissionregistrationV1().MutatingWebhookConfigurations().Get(context.TODO(), mutatingWebhookConfiguration, metav1.GetOptions{})
+	if err != nil {
+		return "", err
+	}
+	for _, webhook := range mutatingWebhook.Webhooks {
+		if webhook.Name == webhookName {
+			for _, matchExpression := range webhook.NamespaceSelector.MatchExpressions {
+				optInLabelSelectorRequirement := metav1.LabelSelectorRequirement{Key: webhookName, Operator: "In", Values: []string{"allocate"}}
```

Comment:
Can we compare with content from patches ? https://github.com/k8snetworkplumbingwg/kubemacpool/blob/master/config/default/mutatepods_opt_in_patch.yaml

so we don't duplicate code.

Reply:
you mean I should parse `mutatevirtualmachines_opt_mode_patch.yaml` and extract the opt-mode from there
thing is - KMP repo runs using `test` folder  `config/test/mutatevirtualmachines_opt_mode_patch.yaml` while CNAO repo runs using `release` folder `config/release/mutatevirtualmachines_opt_mode_patch.yaml`

so it's not comparable without knowing what's the repo from which the test is run.. I can check by checking the namespace, but it's too messy so I think we should stick to the current approach

Reply:
I would say that those patches belongs to a common place the don't differ from test and release, but maybe not at this patch.

Reply:
agreed, I initially tried to put them in the pace to avoid duplicate patched but kustomize was stubborn, It didn't even eat a symbolic link pointing to a common file..
Let's try to tackle it in a different PR..